lib/beaker/hypervisor/google_compute.rb

```diff
@@ -70,138 +86,141 @@ def initialize(google_hosts, options)
+          # take too much work on beaker to add sudo support to everything
+          {
+            key: 'enable-oslogin',
+            value: 'FALSE'
+          },
+        ]
+        next if mdata.empty?
+        # Add the metadata to the host
+        @gce_helper.set_metadata_on_instance(host['vmhostname'], mdata)
+        @logger.debug("Added tags to Google Compute instance #{host.name}: #{host['vmhostname']}")
+
+        host['ip'] = instance.network_interfaces[0].access_configs[0].nat_ip
+
+        # Add the new host to the firewall
+        @gce_helper.add_firewall_tag(@firewall, host['vmhostname'])
+
+        if host['disable_root_ssh'] == true
+          @logger.info('Not enabling root ssh as disable_root_ssh is true')
+        else
 
-        # get ip for this host
-        host['ip'] = instance['networkInterfaces'][0]['accessConfigs'][0]['natIP']
+          # # configure ssh
+          default_user = host['user']
 
-        # configure ssh
-        default_user = host['user']
-        host['user'] = 'google_compute'
+          # TODO: Pull this out into a configuration option or something
+          host['user'] = 'google_compute'
 
-        copy_ssh_to_root(host, @options)
-        enable_root_login(host, @options)
-        host['user'] = default_user
+          # Set the ssh private key we need to use
+          host.options['ssh']['keys'] = [find_google_ssh_private_key]
 
-        # shut down connection, will reconnect on next exec
-        host.close
+          copy_ssh_to_root(host, @options)
+          enable_root_login(host, @options)
+          host['user'] = default_user
+
+          # shut down connection, will reconnect on next exec
+          host.close
+        end
 
         @logger.debug("Instance ready: #{host['vmhostname']} for #{host.name}}")
       end
     end
 
     # Shutdown and destroy virtual machines in the Google Compute Engine,
     # including their associated disks and firewall rules
-    def cleanup()
-      attempts = @options[:timeout].to_i / SLEEPWAIT
-      start = Time.now
-
-      @gce_helper.delete_firewall(@firewall, start, attempts)
+    def cleanup
+      @gce_helper.delete_firewall(@firewall)
 
       @hosts.each do |host|
-        @gce_helper.delete_instance(host['vmhostname'], start, attempts)
+        # TODO: Delete any other disks attached during the instance creation
+        @gce_helper.delete_instance(host['vmhostname'])
         @logger.debug("Deleted Google Compute instance #{host['vmhostname']} for #{host.name}")
-        @gce_helper.delete_disk(host['diskname'], start, attempts)
-        @logger.debug("Deleted Google Compute disk #{host['diskname']} for #{host.name}")
-      end
-
-    end
-
-    # Shutdown and destroy Google Compute instances (including their associated
-    # disks and firewall rules) that have been alive longer than ZOMBIE hours.
-    def kill_zombies(max_age = ZOMBIE)
```

Comment:
Was it verified that this is no longer necessary?

Previously, if you killed a beaker job mid-flight it would leave all sorts of garbage around.

Reply:
I can't find any way in the in beaker code that it could be called. If you can point me to where it can be called, and/or where it's documented, I'll be happy to add it back.

Reply:
Documentation added about manual cleanup.